### PR TITLE
Fix log levels and enable ethereum-go logger

### PR DESCRIPTION
**Pull request by divan**
This PR addresses https://github.com/status-im/status-go/issues/255 (turn on `ethereum-go` log) and fixes log levelling.

In current version of `develop` branch, command line flag `--log` doesn't affect log output - it outputs always at INFO level. Part of this PR fixes this moment.

A few notes:
 - status-go uses two logger variables (own and `ethereum-go`), both are instances of `github.com/ethereum/go-ethereum/log.(Logger)` types, which is, in fact, a fork of [log15 log package](https://github.com/inconshreveable/log15)
 - now both loggers initialized in `geth/log.go` with given log level
 - `status-go` and `github.com/ethereum/go-ethereum/log` use different different definitions for log levels ("ERROR"/"DEBUG"/etc vs "error"/"debug"/etc), so some conversion is being made during log initialization.
 - as `statusd` is both command line app and server, it makes sense to initialize logging only for server mode. It is true for now, but not explicitly, because `log.Init()` is called in `makeNodeConfig()` function. Probably makes sense to make this moment more visible.

### CODE REVIEW

**Comment by jarradh** [comment]
Thank-you very much for your contribution @divan !

**Comment by divan** [comment]
@tiabc thanks for clarifying about tests, I've missed that moment.

I totally agree that tests should have error log level enabled, but what I'm not sure about is that "log level" is a property of the Node (part of NodeConfig, in other words), and not the program itself.

If logging level/options is defined in the scope of the node, then it should've been possible to start many nodes in the program, and defined different log levels for each node. In this case defining log as global `logger` variable doesn't make sense, and each node should keep it's own copy of logger as a dependency).

Perhaps, I don't see the whole picture of what is going on in the statusd server (and future plans) yet, but If we don't need "different log levels per node" case, then logger should be defined at the scope of the whole program, not the individual node. It's much simpler and clearer. For testing, then, we just need to set the desired level for the global logger ("ERROR").

Bottom line, I think we should choose between "logger per node" and "one logger per program". What do you think?

PS. I also was thinking to propose even more drastic simplification of logging, as "TRACE", "WARN", etc level is sort of relict of pre-metrics/logging era, and normally we just need normal/error log output. I tried to capture this idea in [WakeMeUpInTheMiddleOfTheNight log level](https://divan.github.io/posts/wakemeupinthemiddleofthenight/) article. Would be interested to hear your thoughts on this :)

**Comment by tiabc** [comment]
Well, I believe there's no such thing as program-level something (except for packages and the main function, of course) or at least there should be as few as possible of those. Log level in my opinion is the configuration of the instantiated logger. Different program components may have different loggers with different log levels passed as a dependency.

However, in simple logging scenarios it's often singleton just for simplicity and clarity like you've correctly noted. And this is exactly our case.

I wouldn't say having log level inside `NodeConfig` means that there may be many `NodeConfig`s and many `Node`s (although it may be interpreted so). It's about separating concerns. I generally disagree that having global objects implies there may be only 1 object per program and having structs implies that there may be many of every one. I think global objects is a poor design decision and everything should be encapsulated in well-defined scopes of responsibilities and composed for the program execution and it has nothing to do with guesses how many of each object can be there in the program.

A future improvement of that is to turn off go-ethereum logs (setting them to ERROR) and only leave StatusIM logs (setting them to INFO), it's sometimes helpful for debugging tests.

ps. Yes, I've read a lot about that and absolutely agree. I'm happy to find a person who also shares this idea :) We may end up there.

**Comment by divan** [comment]
@tiabc thanks for clarifying the problem with testing and pointing to general refactoring issue.

> Summarily, I wouldn't change log initialising and leave it at package init level, not in a separate function
I think `log.Init()` was a wrong choice of name, as, in fact, this function just sets the level and instantiates log handlers (including go-ethereum one). I renamed it to `SetLevel()` and added explicit default log level setting in `init()` just to avoid having nil logger.

So, in current implementation, logger is still global, and by default is initialized with level "INFO". During the node config initialization, it gets updated via `log.SetLevel()` call. The same applies to the testing - in `init()` function the `log.SetLevel("ERROR")` call explicitly sets logging level to the "ERROR".

One caveat - LogLevel in NodeConfig still seems redundant. Maybe it's worth to keep in mind for future code refactor.

@tiabc does it look good to you?

**Comment by tiabc** [comment]
Thank you for the contribution! I still have some comments but they can be left until next refactoring. Looking forward to your involvement again.
